torch_geometric/datasets/movie_lens_100k.py

```diff
@@ -0,0 +1,133 @@
+    (1000 nodes). User ratings for movies are available as ground truth labels
+    for the edges between the users and the movies
+    :obj:`("user", "rates", "movie")`.
+
+    Args:
+        root (str): Root directory where the dataset should be saved.
+        split (str, optional): If :obj:`"train"`, loads the training dataset.
+            If :obj:`"test"`, loads the test dataset. (default: :obj:`"train"`)
+        transform (callable, optional): A function/transform that takes in an
+            :obj:`torch_geometric.data.HeteroData` object and returns a
+            transformed version. The data object will be transformed before
+            every access. (default: :obj:`None`)
+        pre_transform (callable, optional): A function/transform that takes in
+            an :obj:`torch_geometric.data.HeteroData` object and returns a
+            transformed version. The data object will be transformed before
+            being saved to disk. (default: :obj:`None`)
+    """
+
+    url = 'https://files.grouplens.org/datasets/movielens/ml-100k.zip'
+
+    def __init__(self, root, split="train",
+                 transform: Optional[Callable] = None,
+                 pre_transform: Optional[Callable] = None):
+        super().__init__(root, transform, pre_transform)
+
+        if split not in {'train', 'test'}:
+            raise ValueError(f"Invalid 'split' argument (got {split})")
+
+        path = self.processed_paths[['train', 'test'].index(split)]
+        self.data, self.slices = torch.load(path)
+
+    @property
+    def raw_file_names(self) -> List[str]:
+        return [
+            osp.join('ml-100k', 'u.item'),
+            osp.join('ml-100k', 'u.user'),
+            osp.join('ml-100k', 'u1.base'),
+            osp.join('ml-100k', 'u1.test'),
+        ]
+
+    @property
+    def processed_file_names(self) -> List[str]:
+        return ['train_data.pt', 'test_data.pt']
+
+    def download(self):
+        path = download_url(self.url, self.raw_dir)
+        extract_zip(path, self.raw_dir)
+        os.remove(path)
+
+    def process(self):
+        import pandas as pd
+
+        # Process movie data
+        df = pd.read_csv(self.raw_paths[0], sep="|", header=None,
+                         names=MOVIE_HEADERS, index_col='movieId',
+                         encoding="ISO-8859-1")
+        movie_mapping = {idx: i for i, idx in enumerate(df.index)}
+
+        genre_headers = MOVIE_HEADERS[6:]
+        movie_features = df[genre_headers].values
+        movie_features = torch.from_numpy(movie_features).to(torch.float)
+
+        # Process user data
+        df = pd.read_csv(
+            self.raw_paths[1], sep="|", header=None,
+            names=["userId", "age", "gender", "occupation",
+                   "zipCode"], index_col='userId', encoding="ISO-8859-1")
+        user_mapping = {idx: i for i, idx in enumerate(df.index)}
+
+        max_age = df["age"].values.max()
+        age = df["age"].values / max_age

```

Comment:
Why do we need to normalize `age`? Isn't it a category feature?

Reply:
ML-100K's `age` column is numeric, whereas ML-1M's `age` column is categorical. Initially, I considered binning it for consistency, but ultimately I opted for the normalization approach used in the [reference design](https://github.com/muhanzhang/IGMC/blob/5fa0e3cb1bdc46ffcb97fa6e0249a553e2e0b438/preprocessing.py#L516).